runtime/avn/src/lib.rs

```diff
@@ -208,16 +210,26 @@ impl frame_support::traits::OnRuntimeUpgrade for SeedBridgeTransactionAndDropEth
 
     #[cfg(feature = "try-runtime")]
     fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
-        let pre_upgrade_transaction_id: u64 = get_nonce_seed().unwrap();
-        Ok((pre_upgrade_transaction_id + 1u64).encode())
+        let onchain = pallet_eth_bridge::Pallet::<Runtime>::on_chain_storage_version();
+
+        if onchain < 1 {
+            let pre_upgrade_transaction_id: u64 = get_nonce_seed().unwrap();
+            return Ok((pre_upgrade_transaction_id + 1u64).encode())
+        }
+
+        Ok(vec![])
     }
 
     #[cfg(feature = "try-runtime")]
     fn post_upgrade(new_transaction_id_bytes: Vec<u8>) -> Result<(), &'static str> {
-        let next_tx_id: u32 = pallet_eth_bridge::Pallet::<Runtime>::get_next_tx_id();
-        let expiry = <pallet_eth_bridge::Pallet<Runtime> as pallet_eth_bridge::Store>::EthTxLifetimeSecs::get();
-        assert_eq!((next_tx_id as u64).encode(), new_transaction_id_bytes);
-        assert_eq!(expiry, 7200);
+        let current = pallet_eth_bridge::Pallet::<Runtime>::current_storage_version();
+
+        if current == 1 {
+            let next_tx_id: u32 = pallet_eth_bridge::Pallet::<Runtime>::get_next_tx_id();
+            let expiry = <pallet_eth_bridge::Pallet<Runtime> as pallet_eth_bridge::Store>::EthTxLifetimeSecs::get();
+            assert_eq!((next_tx_id as u64).encode(), new_transaction_id_bytes);
+            assert_eq!(expiry, 7200);
+        }
```

Comment:
can we discuss this offline? Not that familiar with the pre/post upgrade of the try-runtime.

Reply:
Sure, more details here: https://github.com/AventusProtocolFoundation/avn-parachain#storage-migration-tests